openmc/material.py

```diff
@@ -560,6 +560,73 @@ def add_element(self, element, percent, percent_type='ao', enrichment=None):
         for nuclide in element.expand(percent, percent_type, enrichment):
             self.add_nuclide(*nuclide)
 
+    def add_material(self, material, percent, percent_type='ao'):
+        """Mix a material to the material
+        Parameters
+        ----------
+        material : openmc.Material
+            Material to add
+        percent : float
+            Atom or weight percent
+        percent_type : {'ao', 'wo'}
+            'ao' for atom percent and 'wo' for weight percent
+        """
+
+        if self._macroscopic is not None:
+            msg = 'Unable to add a Material to Material ID="{0}" as a ' \
+                  'macroscopic data-set has already been added'.format(self.id)
+            raise ValueError(msg)
+
+        if not isinstance(material, openmc.Material):
+            msg = 'Unable to add a Material to Material ID="{0}" with a ' \
+                  'non-Material value "{1}"'.format(self.id, material)
+            raise ValueError(msg)
+
+        elif not isinstance(percent, Real):
+            msg = 'Unable to add a Material to Material ID="{0}" with a ' \
+                  'non-floating point value "{1}"'.format(self.id, percent)
+            raise ValueError(msg)
+
+        elif self.temperature != material.temperature:
+            msg = 'Unable to add a Material to Material ID="{0}" with a ' \
+                  'different temperature "{1}"'.format(self.id, temperature)
+            raise ValueError(msg)
+
+        elif percent_type not in ['ao', 'wo']:
+            msg = 'Unable to add a Material to Material ID="{0}" with a ' \
+                  'percent type "{1}"'.format(self.id, percent_type)
+            raise ValueError(msg)
+
+        self_percent = 1. - percent
+
+        # Get the atom densities in the given material
+        new_nucs = {}
+
+        # Compute the sum of the atom densities
+        amm = material.average_molar_mass
+        density_sum = 0.
+        for nuc, dens in material.get_nuclide_atom_densities().values():
+            # Convert to weight percent, if requested
+            if percent_type == 'wo':
+                if isinstance(nuclide, openmc.Nuclide):
+                    atomic_mass = openmc.data.atomic_mass(nuclide.name)
+                else:
+                    atomic_mass = openmc.data.atomic_mass(nuclide)
+
+                newdens = dens * atomic_mass / amm
+
+            new_nucs[nuc] = (nuc, newdens)
+            density_sum += newdens
+
+        # Add the nuclide to the material
+        for nuc, dens in new_nucs.values():
+            density = dens / density_sum
+            self.add_nuclide(nuc, density*percent, percent_type)
```

Comment:
If the nuclide is already present in the material, this will result in the material having that nuclide duplicated. It would be better to modify the density of the existing nuclide.